rationale.md

```diff
@@ -383,42 +427,192 @@ async create_query_task(
+  /**
+   * Force use of production models even if the user is in development mode.
+   */
   force_production?: boolean
-  // Retrieve any results from cache even if the results have expired.
+  /**
+   * Retrieve any results from cache even if the results have expired.
+   */
   cache_only?: boolean
-  // Prefix to use for drill links (url encoded).
+  /**
+   * Prefix to use for drill links (url encoded).
+   */
   path_prefix?: string
-  // Rebuild PDTS used in query.
+  /**
+   * Rebuild PDTS used in query.
+   */
   rebuild_pdts?: boolean
-  // Perform table calculations on query results
+  /**
+   * Perform table calculations on query results
+   */
   server_table_calcs?: boolean
-  // Requested fields
+  /**
+   * Requested fields
+   */
   fields?: string
 }
 ```
 
-### Missing features
+#### Methods with delimited list types
+
+As indicated in the [style values section of the OpenAPI specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#style-values), a list of values can be provided to a REST call in a variety of delimited formats. The Looker API uses "csv" formatting for some requests that receive simple lists, such as IDs.
+
+In Typescript, calling `toString()` on an array of numbers such as `[1,2,3]` would return `1, 2, 3`, which is fine for the Looker API methods expecting "csv-formatted" lists. However, in Python, the equivalent function returns `[1, 2, 3]` which would cause a parsing error by the Looker API endpoint.
+
+As indicated in the OpenAPI specification style values, other delimiters such as pipe (`|`), space, or tab characters could be expected by an API endpoint.
+
+To guarantee consistency in the serialization of "csv-formatted" array values, the code generator uses a **Delim** type for parameters that are specified as having a special delimiter. In Typescript, this type is called `DelimArray`. In Python, it's called `DelimSequence`.
+Using this type for the parameter also helps document the expected format for the list submitted to the API endpoint.
+
+Both `DelimArray` and `DelimSequence` accept optional overrides for:
+
+- `separator`, which default to a comma. This can be overridden with pipe, space, tab, tilde or whatever string separator might be required.
+
+- `prefix`, which defaults to an empty string.
+
+- `suffix`, which defaults to an empty string.
+
+### Binary response handling
+
+The popular request libraries currently used by the RTLs for both Python and the Node-based Typescript SDK treat response content as string by default, with binary mode as an opt-in setting. This results in the download of binary content like images or PDFs from SDK methods to be corrupted.
+
+Until we've [resolved this issue](#binary-response-detection) with replacement HTTP request methods, the code generator will assist with binary responses in the following ways (the provided examples in Typescript).
+
+#### Response can be either binary and string
```

Comment:
What are your thoughts about also showing the python equivalent of the snippets below? We won't have to do it for all languages we support but given that Python is the most popular language amongst our customers it might be worth considering (once we implement this support for the Python SDK at least.)

Reply:
After discussing with @joeldodge79 and @dthorpe I think we can completely hide the `BinaryMode` flag requirement from the user. Still working on rewriting what I did this weekend for that. The comments in the method header will still be worth producing.